python/llm/src/bigdl/llm/transformers/models/chatglm2.py

```diff
@@ -370,9 +371,13 @@ def core_attn_forward_8eb45c(self, query_layer, key_layer, value_layer, attentio
                                                                              value_layer,
                                                                              attention_mask,
                                                                              is_causal=True)
+        elif attention_mask is None:
+            scaling_factor = 1 / math.sqrt(query_layer.size(-1))
+            attn = torch.matmul(query_layer * scaling_factor, key_layer.transpose(-2, -1))
```

Comment:
Does this optimization can be used for first & rest token or only for rest token?
If only for rest token, shall we add limitation such as `elif attention_mask is None and bs == 1` ?

Reply:
> Does this optimization can be used for first & rest token or only for rest token?

not tried yet, may try 1st token later.

> If only for rest token, shall we add limitation such as `elif attention_mask is None and bs == 1`

I think the condition `query_layer.shape[2] == key_layer.shape[2]` is enough.